src/main/java/org/openstreetmap/atlas/generator/AtlasGenerator.java

```diff
@@ -284,243 +360,428 @@ public void start(final CommandMap command)
         final List<AtlasGenerationTask> tasks = generateTasks(countries, boundaries, sharding);
         logger.debug("Generated {} tasks in {}.", tasks.size(), timer.elapsedSince());
 
-        // Transform the map country name to shard to country name to Atlas
-        // This is not enforced, but it has to be a 1-1 mapping here.
-        final Map<String, String> configurationMap = configurationMap();
-        final JavaPairRDD<String, Atlas> countryAtlasShardsRDD = getContext()
+        if (useRawAtlas)
+        {
+            // AtlasLoadingOption isn't serializable, neither is command map. To avoid duplicating
+            // boiler-plate code for creating the AtlasLoadingOption, extract the properties we need
+            // from the command map and pass those around to create the AtlasLoadingOption
+            final Map<String, String> atlasLoadingOptions = extractAtlasLoadingProperties(command);
+
+            // Generate the raw Atlas and save it
+            final JavaPairRDD<String, Atlas> countryRawAtlasShardsRDD = generateAndSaveRawAtlas(
+                    tasks, boundaries, sparkContext, atlasLoadingOptions, pbfContext, output,
+                    codeVersion, dataVersion);
+
+            // Slice the raw Atlas and save result
+            final JavaPairRDD<String, Atlas> countrySlicedRawAtlasShardsRDD = sliceAndSaveRawAtlases(
```

Comment:
To make it easier to read the Spark flow, it would be nicer to see the "mapToPair" "filter", "reduce" calls here directly.

What I would do instead here is make the `generateAndSaveRawAtlas` method return a `Function` that does the transformation for you between the `Tuples`. That way all you have to do here is something that will look like this:
```
final JavaPairRDD<String, Atlas> countrySlicedRawAtlasShardsRDD = 
    countryRawAtlasShardsRDD.mapToPair(sliceAndSaveRawAtlases(.....));
```

Reply:
Fair point, I'll update this!